### ISSUE TITLE
rlwrap error

### ISSUE THREAD

**Issue by adslibre** [sritchie/mit-scheme-docker#1]
When I try to run the container, it returns the following error and stop the container.

`$ docker run --interactive --tty sritchie/mechanics`
`rlwrap: error: My terminal reports width=0 (is it emacs?)  I can't handle this, sorry!`

My docker version:
                                                                                
>Client: Docker Engine - Community
 Version:           19.03.9
 API version:       1.40
 Go version:        go1.13.10
 Git commit:        9d988398e7
 Built:             Fri May 15 00:22:47 2020
 OS/Arch:           linux/amd64
 Experimental:      false
>
>Server: Docker Engine - Community
 Engine:
  Version:          19.03.9
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.10
  Git commit:       9d988398e7
  Built:            Fri May 15 00:28:17 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683

**Comment by sritchie** [comment]
Strange, can you give this a try?

```
docker run -it --entrypoint mechanics sritchie/mechanics
```
I think the problem is this little trick, where I call `with-rlwrwap`:

https://github.com/sritchie/mit-scheme-docker/blob/master/mechanics/Dockerfile#L22

It's possible that I need to sleep SLIGHTLY longer:

https://github.com/sritchie/mit-scheme-docker/blob/master/mit-scheme/resources/with_rlwrap.sh#L4

You might find this script helpful, btw, instead of calling docker directly:

https://github.com/sritchie/sicm/blob/master/bin/mechanics

I'm curious to see what happens if you add a slight wait to that script. I can help later today. Thanks for the report, and for using the container!

**Comment by adslibre** [comment]
> Strange, can you give this a try?
> 
> ```
> docker run -it --entrypoint mechanics sritchie/mechanics
> ```

It returns the following:
```
$ docker run -it --entrypoint mechanics sritchie/mechanics
MIT/GNU Scheme running under GNU/Linux
Type `^C' (control-C) followed by `H' to obtain information about interrupts.

Copyright (C) 2019 Massachusetts Institute of Technology
This is free software; see the source for copying conditions. There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Image saved on Friday August 30, 2019 at 11:20:36 PM
  Release 10.1.10 || Microcode 15.3                 || Runtime 15.7
  SF 4.41         || LIAR/x86-64 4.118              || SOS 1.8
  XML 1.0         || Edwin 3.117                    || X11 1.3
  X11-Screen 1.0  || ScmUtils Mechanics.Summer 2019

1 ]=> 
```

> I think the problem is this little trick, where I call `with-rlwrwap`:
> 
> https://github.com/sritchie/mit-scheme-docker/blob/master/mechanics/Dockerfile#L22

Yes, I think so too.

**Comment by adslibre** [comment]
> It's possible that I need to sleep SLIGHTLY longer:
> 
> https://github.com/sritchie/mit-scheme-docker/blob/master/mit-scheme/resources/with_rlwrap.sh#L4
> (...)
> 
> I'm curious to see what happens if you add a slight wait to that script. I can help later today. Thanks for the report, and for using the container!

It worked for me after I changed the value of the sleep parameter for 0.2 in the script.

**Comment by sritchie** [comment]
Okay @asmLibre , I've just merged #2 , which ups the sleep to 0.2. Dockerhub should build the new container shortly. Thanks again for the report!

I'm curious what you're up to with mechanics. Are you working through SICM? I'd love to hear what you're trying to do, in case I can help make it easier.

Cheers,
Sam
